feature/milestone details UI

mshankarrao opened the pull request.

samelamin commented on api/src/frontend/components/milestoneItem.tsx:
```diff
@@ -1,25 +1,61 @@
+  isUserInitiator: boolean
 }
 
+// const SecondFields = () => (
```
Can we remove the commented out code please

samelamin commented on api/src/frontend/components/milestoneItem.tsx:
```diff
@@ -56,6 +98,21 @@ class MilestoneItem extends React.Component<MilestonesItemProps, MilestonesState
     }
   }
 
+   displayDetailsField(): JSX.Element {
```
We want to display details to everyone? 

Should this be called EditDetails?

mshankarrao replied:
Yeah we are displaying to everyone and the button appears only if it is the initiator

mshankarrao commented:
Updated the code with latest main changes. Added the code for show the details to all the users and add to the initiator.
Added exception handling wherever necessary. @samelamin please review and let me know your comments

samelamin commented:
LGTM, @cuteolaf please review when you get a chance

cuteolaf reviewed (commented):
good work, @mshankarrao !

cuteolaf commented:
https://github.com/ImbueNetwork/dapp/blob/98b205a9a3d118d96c7cdea352cef02a98301076/api/src/backend/routes/api/v1/users.ts#L16-L24

I don't think this is correct.

https://github.com/ImbueNetwork/dapp/blob/98b205a9a3d118d96c7cdea352cef02a98301076/api/src/backend/routes/api/v1/users.ts#L23

mshankarrao commented:
> good work, @mshankarrao !

Thanks buddy

mshankarrao commented:
> @mshankarrao
> 
> https://github.com/ImbueNetwork/dapp/blob/98b205a9a3d118d96c7cdea352cef02a98301076/api/src/backend/routes/api/v1/users.ts#L16-L24
> 
> I don't think this is correct.
> 
> https://github.com/ImbueNetwork/dapp/blob/98b205a9a3d118d96c7cdea352cef02a98301076/api/src/backend/routes/api/v1/users.ts#L23

This i haven't change @cuteolaf

cuteolaf commented:
> @mshankarrao
> 
> https://github.com/ImbueNetwork/dapp/blob/98b205a9a3d118d96c7cdea352cef02a98301076/api/src/backend/routes/api/v1/users.ts#L16-L24
> 
> I don't think this is correct.
> 
> https://github.com/ImbueNetwork/dapp/blob/98b205a9a3d118d96c7cdea352cef02a98301076/api/src/backend/routes/api/v1/users.ts#L23

Let me know what you think about the it. @mshankarrao @samelamin

samelamin commented on api/src/backend/db/migrations/20221024100246_milestone.ts:
```diff
@@ -5,21 +5,26 @@ import { auditFields, onUpdateTrigger } from "../utils";
-    const milestoneDetailsTableName = "milestone_details";
-    await knex.schema.createTable(milestoneDetailsTableName, (builder) => {
-        builder.integer("index").notNullable();
+    const milestonesTableName = "milestones";
```
I think this migration will fail because the milestone table already exists

You should have added a migration that on `up` only adds the column and on `down` deletes the column

I will merge it in now because we are migrating AWS environments so this isnt an issue since we need to create the DB from the start and we are lucky we dont have data populated on prod that we cant recreate

mshankarrao replied:
you mean alter table, i can update that if you want

samelamin replied:
yes please, it should be altering an existing table since its a new migration not the first one
